### PR TITLE
remove sdk error

### PR DESCRIPTION
**Pull request by ThanhNhann**
Closes: #371

## Context and purpose of the change

This pull request replaces sdkerrors with regular go errors for all modules of stride

## Author's Checklist

I have...

- [ ] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [ ] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

**Comment by faddat** [comment]
not needed for v4

File: x/stakeibc/keeper/icacallbacks_claim.go [hunk 1]
```diff
@@ -56,7 +55,7 @@ func ClaimCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, ack *c
 	txMsgData, err := icacallbacks.GetTxMsgData(ctx, *ack, k.Logger(ctx))
 	if err != nil {
 		k.Logger(ctx).Error(fmt.Sprintf("failed to unmarshal txMsgData, packet %v", packet))
-		return sdkerrors.Wrap(icacallbackstypes.ErrTxMsgData, err.Error())
+		return fmt.Errorf(icacallbackstypes.ErrTxMsgData.Error(), err.Error())
```
Comment (ThanhNhann) [inline x/stakeibc/keeper/icacallbacks_claim.go:58]: @hoangdv2429: It's should be `return fmt.Errorf("%s: %s", err.Error(), icacallbackstypes.ErrTxMsgData.Error())`

File: x/stakeibc/keeper/icacallbacks_delegate.go [hunk 2]
```diff
@@ -63,7 +62,7 @@ func DelegateCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, ack
 	txMsgData, err := icacallbacks.GetTxMsgData(ctx, *ack, k.Logger(ctx))
 	if err != nil {
 		k.Logger(ctx).Error(fmt.Sprintf("failed to fetch txMsgData, packet %v", packet))
-		return sdkerrors.Wrap(icacallbackstypes.ErrTxMsgData, err.Error())
+		return fmt.Errorf(icacallbackstypes.ErrTxMsgData.Error(), err.Error())
 	}
```
Comment (ThanhNhann) [inline x/stakeibc/keeper/icacallbacks_delegate.go:66]: @hoangdv2429: I think it should be: `return fmt.Errorf("%s: %s", err.Error(), icacallbackstypes.ErrTxMsgData.Error())`

File: x/stakeibc/keeper/icacallbacks_rebalance.go [hunk 3]
```diff
@@ -42,7 +41,7 @@ func RebalanceCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, ac
 	txMsgData, err := icacallbacks.GetTxMsgData(ctx, *ack, k.Logger(ctx))
 	if err != nil {
 		k.Logger(ctx).Error(fmt.Sprintf("failed to fetch txMsgData, packet %v", packet))
-		return sdkerrors.Wrap(icacallbackstypes.ErrTxMsgData, err.Error())
+		return fmt.Errorf(icacallbackstypes.ErrTxMsgData.Error(), err.Error())
```
Comment (ThanhNhann) [inline x/stakeibc/keeper/icacallbacks_rebalance.go:44]: @hoangdv2429 I think it should be `return fmt.Errorf("%s: %s", err.Error(), icacallbackstypes.ErrTxMsgData.Error())`

File: x/stakeibc/keeper/icacallbacks_redemption.go [hunk 4]
```diff
@@ -64,7 +63,7 @@ func RedemptionCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, a
 	txMsgData, err := icacallbacks.GetTxMsgData(ctx, *ack, k.Logger(ctx))
 	if err != nil {
 		k.Logger(ctx).Error(fmt.Sprintf("RedemptionCallback txMsgData could not be parsed, packet %v", packet))
-		return sdkerrors.Wrap(icacallbackstypes.ErrTxMsgData, err.Error())
+		return fmt.Errorf(icacallbackstypes.ErrTxMsgData.Error(), err.Error())
```
Comment (ThanhNhann) [inline x/stakeibc/keeper/icacallbacks_redemption.go:66]: @hoangdv2429 plz update this like format I have cmted

File: x/stakeibc/keeper/icacallbacks_reinvest.go [hunk 5]
```diff
@@ -44,7 +43,7 @@ func ReinvestCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, ack
 	txMsgData, err := icacallbacks.GetTxMsgData(ctx, *ack, k.Logger(ctx))
 	if err != nil {
 		k.Logger(ctx).Error(fmt.Sprintf("failed to fetch txMsgData, packet %v", packet))
-		return sdkerrors.Wrap(icacallbackstypes.ErrTxMsgData, err.Error())
+		return fmt.Errorf(icacallbackstypes.ErrTxMsgData.Error(), err.Error())
```
Comment (ThanhNhann) [inline x/stakeibc/keeper/icacallbacks_reinvest.go:46]: @hoangdv2429 also this one

File: x/stakeibc/keeper/icacallbacks_undelegate.go [hunk 6]
```diff
@@ -68,7 +67,7 @@ func UndelegateCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, a
 	txMsgData, err := icacallbacks.GetTxMsgData(ctx, *ack, k.Logger(ctx))
 	if err != nil {
 		k.Logger(ctx).Error(fmt.Sprintf("failed to fetch txMsgData, packet %v", packet))
-		return sdkerrors.Wrap(icacallbackstypes.ErrTxMsgData, err.Error())
+		return fmt.Errorf(icacallbackstypes.ErrTxMsgData.Error(), err.Error())
```
Comment (ThanhNhann) [inline x/stakeibc/keeper/icacallbacks_undelegate.go:70]: @hoangdv2429 also this one

File: x/stakeibc/module_ibc.go [hunk 7]
```diff
@@ -159,7 +158,7 @@ func (im IBCModule) OnAcknowledgementPacket(
 		errMsg := fmt.Sprintf("Unable to call registered callback from stakeibc OnAcknowledgePacket | Sequence %d, from %s %s, to %s %s",
 			modulePacket.Sequence, modulePacket.SourceChannel, modulePacket.SourcePort, modulePacket.DestinationChannel, modulePacket.DestinationPort)
 		im.keeper.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(icacallbacktypes.ErrCallbackFailed, errMsg)
+		return fmt.Errorf(icacallbacktypes.ErrCallbackFailed.Error(), errMsg)
```
Comment (ThanhNhann) [inline x/stakeibc/module_ibc.go:161]: @hoangdv2429 also this line

File: x/stakeibc/types/message_redeem_stake.go [hunk 8]
```diff
@@ -43,24 +44,24 @@ func (msg *MsgRedeemStake) ValidateBasic() error {
...
 	_, err := sdk.AccAddressFromBech32(msg.Creator)
 	if err != nil {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+		return fmt.Errorf("invalid creator address (%s): invalid address", err.Error())
 	}
 	// validate host zone is not empty
 	// we check validity in the RedeemState function
 	if msg.Receiver == "" {
-		return sdkerrors.Wrapf(ErrRequiredFieldEmpty, "receiver cannot be empty")
+		return fmt.Errorf("receiver cannot be empty:  %s", ErrRequiredFieldEmpty)
```
Comment (ThanhNhann) [inline x/stakeibc/types/message_redeem_stake.go:52]: @hoangdv2429 have one more space between `:` and `%` here

File: cmd/strided/config/config.go [hunk 9]
```diff
@@ -72,18 +73,19 @@ func RegisterDenoms() {
...
 
 		if len(bytes) > address.MaxAddrLen {
-			return sdkerrors.Wrapf(sdkerrors.ErrUnknownAddress, "address max length is %d, got %d", address.MaxAddrLen, len(bytes))
+			return fmt.Errorf("address max length is %d, got %d: unknown address", address.MaxAddrLen, len(bytes))
 		}
 
 		// TODO: Do we want to allow addresses of lengths other than 20 and 32 bytes?
 		if len(bytes) != 20 && len(bytes) != 32 {
-			return sdkerrors.Wrapf(sdkerrors.ErrUnknownAddress, "address length must be 20 or 32 bytes, got %d", len(bytes))
+			return fmt.Errorf("address length must be 20 or 32 bytes, got %d: : unknown address", len(bytes))
```
Comment (ThanhNhann) [inline cmd/strided/config/config.go:88]: @hoangdv2429 plz check this format again

File: utils/utils.go [hunk 10]
```diff
@@ -154,7 +152,7 @@ func AccAddressFromBech32(address string, bech32prefix string) (addr AccAddress,
 	return AccAddress(bz), nil
 }
 
-//==============================  AIRDROP UTILS  ================================
+// ==============================  AIRDROP UTILS  ================================
```
Comment (ThanhNhann) [inline utils/utils.go:155]: @hoangdv2429 plz don't edit this

Comment (faddat) [reply utils/utils.go:155]: if it is happening over and over, it could be an issue with his editor config.

Comment (hoangdv2429) [reply utils/utils.go:155]: @ThanhNhann @faddat I'm really sorry about this. It's indeed related to me editor config, after cltr + s. My editor auto add space to some places (the one above for example). I'll try to fix my editor. I'm really sorry.

**Comment by faddat** [comment]
Ok so, I've reverted my commits and... the annoying formatting thing when you save, it works like -- well, here's how I have my on-save stuff set up:

image


Basically, since we use golangci-lint, it's best to set stuff up like that. 

And worry not!  It's really minor stuff, and this branch looks great.  Thanks!

**Comment by hoangdv2429** [comment]
@faddat Thank you so much! It's definitely my fault for not checking my commit more carefully before pushing. Thanks for your teaching sir!! I changed my 'go lint' setting.

**Pull request closed without merging**